runtime/src/main/java/io/quarkiverse/openapi/generator/codegen/SpecInputModel.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkiverse.openapi.generator.codegen;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.InputStream;
+import java.util.Objects;
+
+public class SpecInputModel {
+
+    private final InputStream inputStream;
+    private final String filename;
+
+    public SpecInputModel(final String filename, final InputStream inputStream) {
+        requireNonNull(inputStream, "InputStream can't be null");
+        requireNonNull(filename, "File name can't be null");
+        this.inputStream = inputStream;
+        this.filename = filename;
+    }
+
+    public String getFileName() {
```

Comment:
why not uri rather than filename?

Reply:
We just need the file name from the origin, which will be used to save it locally. So if we have:\

- http://example.com/myspecA.json -> myspecA.json
- /home/luke/files/myspecB.yaml -> myspecB.yaml
- classpath://files/myspecC.yml -> myspecC.yaml

URI won't be used in this extension since our input model is a stream.

Reply:
How are we going to map a function which operation is  /home/luke/files/myspecB.yaml#doSomething to the classes generated from that URI is the information that we pass to the generation is just myspecA.json?
The answer is by defining a known mapping between URI and file in workflow. That mapping might be tricky 
Imagine we have these two URIS: http://example1.com/myspecA.json and  http://example2.com/myspecA.json, we need to have the whole URI into account when generating the file name. Anyway, this is an issue for the implementor, not this PR